### PR TITLE
dependency: bump pitest-maven to 1.11.0

### PR DESCRIPTION
**Pull request by rnveach**
Bump pitest https://github.com/hcoles/pitest/releases/tag/1.11.0 .

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -222,7 +222,7 @@
     <maven.versions.plugin.version>2.13.0</maven.versions.plugin.version>
     <maven.compiler.plugin.version>3.10.1</maven.compiler.plugin.version>
     <java.version>11</java.version>
-    <pitest.plugin.version>1.9.9</pitest.plugin.version>
+    <pitest.plugin.version>1.11.0</pitest.plugin.version>
```
Comment (rnveach) [inline pom.xml:225]: @romani It looks like something is wrong with dependabot as it has stopped updating this plugin.

Last update https://github.com/checkstyle/checkstyle/pull/12293

https://github.com/hcoles/pitest/releases
We are missing plugins 1.9.10 through 1.10.4

Comment (romani) [reply pom.xml:225]: We can send them feedback to investigate and improve.

We have azure weekend job to remind us on any dependency that dependabot does not see.

Comment (romani) [reply pom.xml:225]: This PR should help a lot in update of suppressions

Comment (rnveach) [reply pom.xml:225]: @romani 
> We have azure weekend job to remind us on any dependency that dependabot does not see.

Can you look to see if it has been reporting pitest? I cleaned up a bunch of my emails and may have removed it.

Last update by dependa was in October. So if azure if reporting, it should have been reporting for 3-4 months.

**Comment by rnveach** [comment]
CI should pass. More updates will be made in https://github.com/checkstyle/checkstyle/issues/12591 to clear out that issue.

File: pom.xml (hunk 1 above)
Comment (romani) [reply pom.xml:225]: I do when I have time, but definetely not case in last several month

**Review by romani — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (rnveach) [reply pom.xml:225]: You provided me the details in PM. New issue clarified why it has not been reporting.

**Pull request merged**